ttnn/cpp/ttnn/operations/conv/conv2d/conv2d_utils.cpp

```diff
@@ -551,15 +510,17 @@ std::tuple<ttnn::Shape, ttnn::MemoryConfig, bool, bool> get_conv_padded_input_sh
             conv_config.transpose_shards ? ShardOrientation::COL_MAJOR : ShardOrientation::ROW_MAJOR;
         ParallelConfig optimal_parallel_config;
         if (is_non_tile_mul_width) {
-            optimal_parallel_config = determine_parallel_config_non_tile_mul_width(
+            optimal_parallel_config = determine_parallel_config(
```

Comment:
this pattern is not great, we call determine_parallel_config in many places, we have this if only in this one place.
Can we push this logic into the method?

Reply:
are you sure that all instances of this method are find w/o that bool like in prepare weights?
Like when we call it to figure out which shard layout is the best?

Reply:
@nkpatel-tt please make sure to check if this bool needs to be set anywhere else.

Reply:
yes I missed those places since these PRs were merged nearby(or at least I rebased). 
Anyway I have added same logic at all places where `determine_parallel_config` is getting called.
@pavlejosipovic  I still am a bit hesitant to add that another variable in determine_parallel_config. Current logic does not look great but It will go away once we have support for all cases for non_tile_mul_width. Please do let me know if you still feel that we have to have that another variable in determine_parallel_config.

Reply:
why is it so bad to do that comapred to this realativly ugly code.
I'm not going to block this PR as this is better than current state, but I would strongly recommend to put this within the function.
You can have a default value for an argument you don't have to pybind it.

Reply:
Done.